### PR TITLE
[Nibe] Code analysis fixes

### PR DESCRIPTION
**Pull request by paulianttila**
Signed-off-by: Pauli Anttila <pauli.anttila@gmail.com>

### CODE REVIEW

**Comment by paulianttila** [comment]
See #3441

File: addons/binding/org.openhab.binding.nibeheatpump/src/main/java/org/openhab/binding/nibeheatpump/internal/connection/SerialConnector.java [hunk 1]
```diff
@@ -228,14 +228,16 @@ public void sendReadMsg() {
                 @Override
                 public void log(String format) {
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.nibeheatpump/src/main/java/org/openhab/binding/nibeheatpump/internal/connection/SerialConnector.java]: Ideally you should just inline it and make it explicit that you do a conditional trace logging, what is the goal of doing it like this? How often do you call it?

**Comment by martinvw** [comment]
Thanks for your quick response!

File: addons/binding/org.openhab.binding.nibeheatpump/src/main/java/org/openhab/binding/nibeheatpump/internal/connection/SerialConnector.java (hunk 1 above)
Comment (wborn) [reply addons/binding/org.openhab.binding.nibeheatpump/src/main/java/org/openhab/binding/nibeheatpump/internal/connection/SerialConnector.java]: I had a look and it's called from the enum `NibeHeatPumpProtocolStates` which would require a static logger which is also against the code style.

Comment (martinvw) [reply addons/binding/org.openhab.binding.nibeheatpump/src/main/java/org/openhab/binding/nibeheatpump/internal/connection/SerialConnector.java]: No its not preferred that is something different than not allowed.

Note that the static logger is a warning and the custom logging strings are an error.

See also:

> As we are in a dynamic OSGi environment, loggers should be non-static, when ever possible and have the name logger.

Comment (wborn) [reply addons/binding/org.openhab.binding.nibeheatpump/src/main/java/org/openhab/binding/nibeheatpump/internal/connection/SerialConnector.java]: Then a fix could be to just remove all log methods from `NibeHeatPumpProtocolContext` and use a static logger in `NibeHeatPumpProtocolStates` instead. :-)

Comment (martinvw) [reply addons/binding/org.openhab.binding.nibeheatpump/src/main/java/org/openhab/binding/nibeheatpump/internal/connection/SerialConnector.java]: :like:

Comment (paulianttila) [reply addons/binding/org.openhab.binding.nibeheatpump/src/main/java/org/openhab/binding/nibeheatpump/internal/connection/SerialConnector.java]: I was using static logger originally, but that was asked to change during code review ;-)

Comment (martinvw) [reply addons/binding/org.openhab.binding.nibeheatpump/src/main/java/org/openhab/binding/nibeheatpump/internal/connection/SerialConnector.java]: You are right, the previous message which I used was to strong, I will use message which matches the coding guidelines from now one, sorry :-)

File: addons/binding/org.openhab.binding.nibeheatpump/src/main/java/org/openhab/binding/nibeheatpump/internal/protocol/NibeHeatPumpProtocolStates.java [hunk 2]
```diff
@@ -26,9 +28,13 @@
         public boolean process(NibeHeatPumpProtocolContext context) {
             if (context.buffer().hasRemaining()) {
                 byte b = context.buffer().get();
-                context.log("Received byte: {}", String.format("%02X", b));
+                if (logger.isTraceEnabled()) {
+                    logger.trace("Received byte: {}", String.format("%02X", b));
+                }
                 if (b == NibeHeatPumpProtocol.FRAME_START_CHAR_FROM_NIBE) {
-                    context.log("Frame start found");
+                    if (logger.isTraceEnabled()) {
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.nibeheatpump/src/main/java/org/openhab/binding/nibeheatpump/internal/protocol/NibeHeatPumpProtocolStates.java]: Note that this check is needed if there is no logic inside the statement, I will create a follow up PR for that later

**Comment by martinvw** [comment]
@paulianttila do you see why Intellij would mention this, I cannot quickly see it:

screen shot 2018-04-05 at 18 56 44

**Comment by martinvw** [comment]
Its from the NibeHeatPumpProtocolStates ENUM

**Pull request merged**